tools/imex-runner/imex-runner.py.in

```diff
@@ -82,6 +118,36 @@ if args.pass_pipeline_file:
 imex_binary_dir = '@IMEX_BINARY_DIR@'
 llvm_binary_dir = '@LLVM_BINARY_DIR@'
 
+def run_pipeline(cmds):
+    """
```

Comment:
why is this needed? Is this to report out failure in the end?

Reply:
Maybe you marked the wrong line but the script returns returncode of the last run command. Yes. It is for reporting failure to "lit". Previously our test command would run imex-runner.py and pipe the result into "FileCheck" And FileCheck's error code was the returncode seen by "lit". Now if FileCheck is run as part of imex-runner.py, "lit" should receive the returncode from "FileCheck".